From 0bd89e485fc035b7e5ac0f5fa3521f99fb77c08b Mon Sep 17 00:00:00 2001
From: wangyan-intel <yan.wang@intel.com>
Date: Thu, 25 May 2023 10:34:01 +0800
Subject: [PATCH] Return MFX_ERR_MORE_SURAFCE for VP9 non-display frame (#7242)

It returns MFX_WRN_DEVICE_BUSY which cause dead loop in sample_decode.
sample_decode doesn't feed one new free surface into RT when receives
MFX_WRN_DEVICE_BUSY like mfx_player.

Signed-off-by: Yan Wang <yan.wang@intel.com>
---
 _studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp b/_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp
index cb659983..37d9c22f 100644
--- a/_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp
+++ b/_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp
@@ -1269,6 +1269,7 @@ mfxStatus VideoDECODEVP9_HW::DecodeFrameCheck(mfxBitstream *bs, mfxFrameSurface1
             if (surface_work) // for model 3 we don't need to zero surface_out because we can alloc next frame inside decoder
             {
                 surface_out = nullptr;
+                sts = MFX_ERR_MORE_SURFACE;
             }
             else // for model 3 if we need more surface we can alloc it inside decoder
             {
-- 
2.34.1

